src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -258,23 +264,53 @@ func (s *service) Bootstrapped(ctx thrift.Context) (*rpc.NodeBootstrappedResult_
 }
 
 func (s *service) Query(tctx thrift.Context, req *rpc.QueryRequest) (*rpc.QueryResult_, error) {
+	var sp opentracing.Span
+	result, sp, err := s.query(tctx, req)
+
+	if sp != nil {
```

Comment:
hm, can you rewire the span creation to give you back a `NoOpSpan` when we're not tracing? that way we'd be able to avoid `if sp != nil` everywhere